internal/util/resource/resource_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"runtime"
+	"runtime/pprof"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type TestTrackObject struct {
+	token *Token
+}
+
+// runGC forces several GC cycles to give the runtime a chance to run cleanups.
+func runGC(t *testing.T) {
+	t.Helper()
+
+	for i := 0; i < 8; i++ {
+		runtime.GC()
+		time.Sleep(10 * time.Millisecond)
+	}
```

Comment:
Why eight iterations? Why 10 milliseconds?

Reply:
Loop cycle is just an random number, sleep time is big enough. There is no deterministic reason.

Reply:
I wonder if just one or maybe two `runtime.GC()` would be reliable, too. If not, we probably could use `runtime.Gosched()`